### DISCUSSION TITLE
Run RequestValidationError before middlewares

### DISCUSSION THREAD

**Discussion by edanbenavi** [fastapi/fastapi#9582] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.exceptions import RequestValidationError
app = FastAPI()

def middleware_example(): 
    pass

async def request_input_validator(request: Request, exc):
    return Exception("Bad Input")


app.add_pre_request_middleware(middleware_example)
app.add_exception_middlewares(RequestValidationError, request_input_validator)
```

### Description

when running this code the "middleware_example" function running before the "request_input_validator"

i want to find an option to use RequestValidationError before the middleware 

The logic behind it :  first check if is the body is according to the pydantic schema 
if the check pass -> move to the middleware 
else : raise error 

### Operating System

macOS

### FastAPI Version

lastest

### Python Version

3.11

**Comment by thesubuck** [comment, marked as answer]
@edanbenavi , You cannot change the order of execution . Middleware runs first before exception handlers (including RequestValidationError). Even you define a custom router and a middleware, the control goes through the app like this : 

> middleware -> CustomRouter -> Route

Since you haven't mentioned a specific middleware, I don't know your use-case to suggest further.
